fix(l10n): missing locales (#992)
pejulian/romcal: docs/localization.md

- `names`: Localized names for specific liturgical days, such as feasts of saints and titles of Our Lord and Our Blessed Mother.
  - Each entry for any name that is added to a specific locale other than `en` must also exist in the `en` locale file, as this is the main source of truth.
  - If a name is not found in the specified locale, romcal will look for it in the fallback locales, ending with `en`.
  - There are also tests to ensure that all names in other locales exist in the `en` locale file. (pending #992)
  - Each entry should have a corresponding src comment with a link to the source of the translation, if applicable.
    - If a reference to a missal is used instead, this is the format we've decided to use:
      - `// src: $sourceName`
        - when the key value can be found exactly in the referenced document
      - `// based on: $sourceName`
        - when the key value is adapted from the referenced document
      - For a given missal or document, the following short form is used: (values are examples)
        - `mr` = Missale Romanum;
        - `fr` = is the locale of the Missal;
        - `2022` = publication year;
        - `ed3` = editio typica tertia;

>         - `angers` = optional suffix when variations exist (e.g. English Missals are published e.g. in the UK, the US, Australia; French Missals are published per diocese in France, but maybe in Canada too).

Reviewer: Just one clarification: the French-language missal is published for several French-speaking countries (including Canada, France, Belgium, Luxembourg, Switzerland, and North Africa). But dioceses also usually publish their own proper missal, which includes the feasts and memorials specific to that diocese. So this is not something unique to dioceses in France, but something I believe all dioceses do.

Other than that LGTM, thank you 👍

Author: sure, that was an example, not so much a rule that only France does this.

Author: exemplification clarified

Reviewer 2: > So this is not something unique to dioceses in France, but something I believe all dioceses do.

I know only about French dioceses which publish diocesan Missals. Maybe there are dioceses in other countries which do so too, however, in Slovakia, the Slovak translation of the Roman Missal (as well the national ordos) includes all the celebrations inscribed to any of the eight Slovak dioceses (probably to save cost, but also there are, I think, up to 20 celebrations inscribed in all of the Slovak dioceses).

Author: i don't think this changes the validity of the example.

- `seasons`: Localized names for liturgical seasons.
  - Uses the format:
    - `season_name`:
      - `season`: Localized name of the season.
      - `weekday`: Localized day format for a given weekday.
      - `sunday`: Localized day format for a given Sunday.
      - `privileged_weekday`: Localized day format for a privileged weekday in the season.
      - Other fields may be present depending on needs of the particular season.
- `periods`: Localized names for liturgical periods.
- `ranks`: Localized names for liturgical ranks.
- `weekdays`: Localized names for the days of the week.
- `months`: Localized names for the months of the year.
- `colors`: Localized names for liturgical colors.
- `names`: Localized names for specific liturgical days, such as feasts of saints and titles of Our Lord and Our Blessed Mother.
  - Each entry for any name that is added to a specific locale other than `en` must also exist in the `en` locale file, as this is the main source of truth.
  - If a name is not found in the specified locale, romcal will look for it in the fallback locales, ending with `en`.

>   - There are also tests to ensure that all names in other locales exist in the `en` locale file. (pending #992)

Reviewer 2: The note is unnecessary, as the PR is this one, therefore, when this PR is merged, the remark will be outdated.

```suggestion
  - There are also tests to ensure that all names in other locales exist in the `en` locale file.
```

>   - Each entry should have a corresponding src comment with a link to the source of the translation, if applicable.

Reviewer 2: ```suggestion
  - Each entry should have a corresponding source comment with a link to the source of the translation, if applicable.
```

    - If a reference to a missal is used instead, this is the format we've decided to use:
      - `// src: $sourceName`
        - when the key value can be found exactly in the referenced document
      - `// based on: $sourceName`
        - when the key value is adapted from the referenced document

>       - For a given missal or document, the following short form is used: (values are examples)

Reviewer 2: Right now, we do this for missals only.

```suggestion
      - For missals, the following short form is used: (values are examples)
```

        - `mr` = Missale Romanum;
        - `fr` = is the locale of the Missal;
        - `2022` = publication year;
        - `ed3` = editio typica tertia;
        - `angers` = optional suffix when variations exist (examples below)
          - English Missals are published e.g. in the UK, the US, Australia
          - French Missals are published per diocese in France, but maybe in Canada too

>       - so this would be the full reference: `// src: mr_fr_2022_ed3_angers`

Reviewer 2: ```suggestion
      - the parts above are joined with an underscore (`_`): `// src: mr_fr_2022_ed3_angers`
```
